sdk/rust/src/ledger_id.rs

```diff
@@ -55,6 +69,66 @@ impl Debug for LedgerId {
 
 impl Display for LedgerId {
     fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
-        f.pad(&base64::encode(&self.0))
+        if self.is_mainnet() {
+            f.write_str("mainnet")
+        } else if self.is_testnet() {
+            f.write_str("testnet")
+        } else if self.is_previewnet() {
+            f.write_str("previewnet")
+        } else {
+            f.write_str(&hex::encode(self.to_bytes()))
```

Comment:
following the above, `to_*` is generally considered to be "not free" so the field access would be the preferred way of doing things in this particular case